cmake/scripts/common/ArchSetup.cmake

```diff
@@ -167,17 +167,6 @@ if(NOT MSVC)
     -Wno-unused-parameter # from -Wextra
   )
 
-  if(CMAKE_COMPILER_IS_GNUCXX)
```

Comment:
I had to look it up because I was wondering why you didn't reuse this. I guess this is deprecated. https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILER_IS_GNUCXX.html

Reply:
Exactly it's deprecated and there are no such variables compilers from other vendors.

Reply:
Probably worth cleaning up the rest in another PR.

```
$ git grep CMAKE_COMPILER_IS_GNUCXX  | wc -l
4
```